### PR TITLE
Add /grantme command

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by paramat** [comment]
This is to avoid having to write your name?

**Comment by red-001** [comment]
Yes, we already have the /giveme command so it would make sense to add a similar command for granting privileges

**Comment by paramat** [comment]
Nice, this will save me typing 'singleplayer' so much.

**Comment by Megaf** [comment]
I support the idea.

**Comment by HybridDog** [comment]
There's a mod for it:
https://github.com/red-001/more_chatcommands/blob/master/init.lua#L12

**Comment by HybridDog** [comment]
Sorry for writing sth you're already aware of…

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -154,53 +154,74 @@ core.register_chatcommand("privs", {
...
 	params = "<name> <privilege>|all",
 	description = "Give privilege to player",
 	func = function(name, param)
-		if not core.check_player_privs(name, {privs=true}) and
-				not core.check_player_privs(name, {basic_privs=true}) then
-			return false, "Your privileges are insufficient."
-		end
 		local grantname, grantprivstr = string.match(param, "([^ ]+) (.+)")
+		print(grantname)
+		print(grantprivstr)
```
Comment (kwolekr) [inline builtin/game/chatcommands.lua]: You forgot to remove these print statements=.

**Comment by sfan5** [comment]
:+1: except the `print()`'s you left in

File: builtin/game/chatcommands.lua (hunk 1 above)
Comment (red-001) [inline builtin/game/chatcommands.lua]: done

**Comment by red-001** [comment]
Removed the `print`'s.

**Comment by Megaf** [comment]
@sfan5 @paramat 
Hey I just got here
And this is crazy
But prints got removed
So merge it maybe

**Comment by paramat** [comment]
I'm just unsure whether it needs another approval.

**Comment by Megaf** [comment]
@paramat I understand. Usually trivial needs only one approval. However, I don't think this is trivial.
But we have one approval from @sfan5, maybe we could have one from you too? Then someone could merge it. Perhaps @kwolekr 
Just have a good look at the code before approving.

**Comment by everamzah** [comment]
Just wanted to state that I think /grant and /give without arguments should assume that the issuer is the receiver.  It seems silly to me that there should be duplicate commands.

**Comment by paramat** [comment]
^ Good idea, worth considering.

**Comment by Ferk** [comment]
@everamzah the problem with that idea is that /grant accepts more than 2 arguments, you can include additional privileges in succession, separated by spaces. Thus if you type "`/grant give fly`" what should it do? give "fly" privilege to a player named "give" or give you "give" and "fly" privileges?

Also, /give and /giveme use itemstrings, and itemstring can have spaces on it, in `/give default:stone 5` I guess it'd be strange (impossible?) for a player to named "default:stone" and for an item to be named "5", but I'd rather keep the logic simple and type 2 characters more for the "me".

**Comment by paramat** [comment]
Yes i agree with Ferk the concept of this PR is fine and more robust, i'm not suggesting we should do the 'no name' thing instead.

**Comment by everamzah** [comment]
Yeah, it's invalid usage to use /give and /grant without arguments anyway!  Nonetheless, it was an aside, but I think possible and worthwhile.  Someday.

(I only suspect that whatever triggers invalid usage would fail to a check whether it would be valid usage if the issuer's player name was assumed.)  (Sorry.)

File: builtin/game/chatcommands.lua [hunk 2]
```diff
@@ -154,53 +154,72 @@ core.register_chatcommand("privs", {
...
-			.. core.privs_to_string(
-				core.get_player_privs(grantname), ' ')
+		end	
+		return handle_grant_command(name, grantname, grantprivstr)
+	end,
+})
+
+core.register_chatcommand("grantme", {
+	params = "<privilege>|all",
+	description = "Grant a privilege to yourself",
```
Comment (paramat) [inline builtin/game/chatcommands.lua]: This allows multiple privs to be granted so 'give privileges to yourself' covers singular and plural.

File: builtin/game/chatcommands.lua [hunk 3]
```diff
@@ -154,53 +154,72 @@ core.register_chatcommand("privs", {
 				core.get_player_privs(param), ' ')
 	end,
 })
+
+local function handle_grant_command(caller, grantname, grantprivstr)
+	if not core.check_player_privs(caller, {privs=true}) and
+			not core.check_player_privs(name, {basic_privs=true}) then
```
Comment (paramat) [inline builtin/game/chatcommands.lua]: 'name' should be 'caller' here?

**Comment by paramat** [comment]
Added line comments.

File: builtin/game/chatcommands.lua (hunk 3 above)
Comment (t4im) [inline builtin/game/chatcommands.lua]: Why not getting rid of `check_player_privs` here, when you're at it :-)

```lua
local privs = minetest.get_player_privs(caller)
if not privs.privilege_to_check and not privs.other_priv_to_check then
end
```

Furthermore i recommend, that `check_player_privs` needs to be deprecated.

File: builtin/game/chatcommands.lua [hunk 4]
```diff
@@ -154,53 +154,72 @@ core.register_chatcommand("privs", {
...
+		end	
+		return handle_grant_command(name, grantname, grantprivstr)
+	end,
+})
+
+core.register_chatcommand("grantme", {
+	params = "<privilege>|all",
+	description = "Grant a privilege to yourself",
+	func = function(name, param)
+		if param == "" then
```
Comment (t4im) [inline builtin/game/chatcommands.lua]: `name` can be `nil`, when called from certain mods (irc delegated for example), which would err in `handle_grant_command` during either form of privilege check. You should check for it here, too.

Comment (red-001) [inline builtin/game/chatcommands.lua]: @t4im name should never be `nil`. If it is then that's a bug in a mod.  Most chatcommands will cause a crash if the name is nil.

Comment (red-001) [inline builtin/game/chatcommands.lua]: @paramat the string.match() in `/grant` used to separate the two parameters, `/grantme` only takes one parameter so it isn't necessary.

Comment (t4im) [inline builtin/game/chatcommands.lua]: Ah, got some facts mixed up, pardon me. There's a border case where looking up connected players can result in `nil`, but that shouldn't apply here, indeed.

File: builtin/game/chatcommands.lua (hunk 3 above)
Comment (red-001) [inline builtin/game/chatcommands.lua]: added

**Comment by red-001** [comment]
Updated.

File: builtin/game/chatcommands.lua (hunk 4 above)
Comment (paramat) [inline builtin/game/chatcommands.lua]: Ok.

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
c31e354342bbcb9a051f0113d41431827efc4435

**Pull request closed without merging**